HDOS-690

pka1024 opened the pull request:
feat(angular 16 update):
- angular 16 update.

Commit:
feat(angular 14 update):
- ng update @angular/core@14 @angular/cli@14 --force.

Commit:
feat(angular 14 update):
- ng update @angular/material@14 --force.

Commit:
feat(angular 14 update):
- ng update @ngrx/store@14.

Commit:
- npm i @danielmoncada/angular-datetime-picker@14.2.0.
- npm i @danielmoncada/angular-datetime-picker-moment-adapter@3.0.1.

Commit:
feat(angular 14 update):
- npm update.

Commit:
feat(angular 14 update):
- Fixed deprecated warning  "~" import in scss.
- Fixed katex no known default import.
- Fixed Angular material expansion-panel.

Commit:
feat(angular 14 update):
- Prettier style changes.
- Fixed circular dependency error, after upgrading "angular-auth-oidc-client".
- Fixed config loading and getting async.

Commit:
feat(angular 14 update):
- Fixed a compatibly issue with
"prettier" >= v3.0.0. It needed "eslint-plugin-prettier" >= v5.0.0 and "eslint-plugin-jsdoc" < v43.0.0 to work or
else linting throws errors in node_modules.

pka1024 committed:
angular 14 update

feat(angular 14 update):
- Updated eslint.
- Fixed unit-tests errors.
- Finished angular-auth-oidc-client migration.

pka1024 committed:
angular 14 update

feat(angular 14 update):
- Switched from "ngx-monaco-editor" to "ngx-monaco-editor-v2" because it's support stopped with angular-v12.

pka1024 committed:
angular 14 update

feat(angular 14 update):
- Switched from "ngx-monaco-editor" to "ngx-monaco-editor-v2" because it's support stopped with angular-v12.

pka1024 committed:
Merge branch 'hetida:release' into feature/angular-update-14

pka1024 committed:
angular 15 update

feat(angular 15 update):
- Updated typescript.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- ng update @angular/core@15 @angular/cli@15 --force.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- ng update @angular/material@15 --force.

pka1024 committed:
angular 15 update

angular 15 update
- ng update @ngrx/store@15.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- npm i angular-auth-oidc-client@15.0.5.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- npm i @danielmoncada/angular-datetime-picker@15.0.2.
- npm i ngx-monaco-editor-v2@15.0.1.
- npm i monaco-editor@0.34.1.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- npm i @angular-eslint/builder@15.2.1.
- npm i @angular-eslint/eslint-plugin@15.2.1.
- npm i @angular-eslint/eslint-plugin-template@15.2.1.
- npm i @angular-eslint/schematics@15.2.1.
- npm i @angular-eslint/template-parser@15.2.1.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- Angular Material 15 migration.

pka1024 committed:
Merge branch 'hetida:release' into HDOS-675

pka1024 committed:
Merge branch 'HDOS-675' of https://github.com/pka1024/hetida-designer into HDOS-675

pka1024 committed:
angular 15 update

feat(angular 15 update):
- Angular Material 15 migration.
- Context-menu fix.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- Updated eslint.

pka1024 committed:
angular 15 update

feat(angular 15 update):
- Updated to node version 16.20.2.
- Updated prettier.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- Updated typescript.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- ng update @angular/core@16 @angular/cli@16 --force.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- ng update @angular/material@16 --force.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- ng update @ngrx/store@16.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- npm i angular-auth-oidc-client@16.0.1.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- npm i @danielmoncada/angular-datetime-picker@16.1.0.
- npm i ngx-monaco-editor-v2@16.0.1.
- npm i monaco-editor@0.41.0.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- npm i @angular-eslint/schematics@16.3.1.
- npm i @angular-eslint/builder@16.3.1.
- npm i @angular-eslint/eslint-plugin@16.3.1.
- npm i @angular-eslint/eslint-plugin-template@16.3.1.
- npm i @angular-eslint/template-parser@16.3.1.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- npm i @typescript-eslint/eslint-plugin@7.7.1".
- npm i @typescript-eslint/parser@7.7.1.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- updated node to version "18.20.2".
- updated eslint to version "8.57.0".
- updated angular-plotly.js to version "5.2.2".
- updated @types/plotly.js to version "2.29.4".

pka1024 committed:
angular 16 update

feat(angular 16 update):
- auth guard migration.
- removed old comment.
- disabled "auth not enabled" button if not enabled.
- angular material migration.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- updated eslint-plugin-jsdoc.

pka1024 committed:
Merge branch 'develop' into HDOS-690

pka1024 committed:
angular 16 update

feat(angular 16 update):
- Updated katex.
- Removed  "preserveSymlinks": true from unit tests config.
- Updated Hetida-Flowchart packages to version 0.7.0.

pka1024 committed:
Merge branch 'develop' into HDOS-690

pka1024 committed:
angular 16 update

feat(angular 16 update):
- added routing.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- added routing.
- fixed query parameter.
- fixed Unit Tests.

thatsmeta commented on frontend/src/app/components/workflow-editor/workflow-editor.component.ts:
```diff
@@ -611,7 +611,10 @@ export class WorkflowEditorComponent {
   }
 
   private _updateWorkflowIfNecessary(): void {
-    if (!this.hasChanges || this.currentWorkflow.state === 'RELEASED') {
```
enum statt string

thatsmeta commented on frontend/src/app/components/protocol-viewer/protocol-viewer.component.ts:
```diff
@@ -147,7 +147,7 @@ export class ProtocolViewerComponent implements AfterViewInit {
     const resultType =
       this.executionResponse.output_types_by_output_name[outputKey];
-    return resultType === IOType.PLOTLYJSON;
+    return resultType === IOType.PLOTLYJSON.toString();
```
toString() entfernen

thatsmeta commented on frontend/src/app/service/http-service/transformation-http.service.ts:
```diff
@@ -21,8 +22,16 @@ export class TransformationHttpService {
 
-  public fetchTransformations(): Observable<Array<Transformation>> {
-    const url = `${this.apiEndpoint}/transformations`;
+  public async fetchTransformations(): Promise<Observable<Transformation[]>> {
```
entfernen, config sollte schon geladen sein

thatsmeta commented on frontend/src/app/service/transformation/transformation.service.ts:
```diff
@@ -112,12 +112,12 @@ export class TransformationService {
-    this.transformationHttpService
-      .fetchTransformations()
-      .subscribe(transformations => {
+  async fetchAllTransformations(): Promise<void> {
```
entfernen

thatsmeta commented on frontend/src/assets/hetida_designer_config.json:
```diff
@@ -1,5 +1,10 @@
   "apiEndpoint": "http://localhost:8080/api",
-  "authEnabled": false,
-  "userInfoText": ""
+  "userInfoText": "",
```
entfernen

thatsmeta commented on frontend/src/index.html:
```diff
@@ -1,5 +1,5 @@
 <!-- eslint-disable prettier/prettier -->
-<!DOCTYPE html>
```
doctype groß

thatsmeta commented on frontend/src/silent-renew.html:
```diff
@@ -1,5 +1,5 @@
 <!-- eslint-disable prettier/prettier -->
-<!DOCTYPE html>
```
doctype groß

thatsmeta commented on frontend/src/style/_typography.scss:
```diff
@@ -1,13 +1,15 @@
 $material-design-icons-font-directory-path: '~material-design-icons-iconfont/dist/fonts/';
-@import '~material-design-icons-iconfont/src/material-design-icons';
+@import 'node_modules/material-design-icons-iconfont/src/material-design-icons';
```
Import ohne node_modules

thatsmeta commented on frontend/src/app/app-routing.module.ts:
```diff
@@ -0,0 +1,27 @@
+import { AppComponent } from './app.component';
+
+const appRoutes: Routes = [
+  { path: '', pathMatch: 'full', redirectTo: 'home' },
```
empty path hinter authcallback verschieben

thatsmeta commented on frontend/src/app/app.component.ts:
```diff
@@ -1,94 +1,15 @@
-    private readonly contextMenuService: ContextMenuService,
-    private readonly authService: AuthService
-  ) {
-    this.iconRegistry.addSvgIcon(
```
iconRegistry Aufruf gehört hierhin und nicht in HomeComponent

thatsmeta commented on frontend/src/app/app.component.ts:
```diff
@@ -1,94 +1,15 @@
+  constructor(private readonly oidcSecurityService: OidcSecurityService) {}
 
   ngOnInit() {
-    this.themeService.currentTheme.subscribe(theme => {
```
themeService Aufruf wenn es geht auch besser hier statt in HomeComponent

thatsmeta commented on frontend/src/app/app.module.ts:
```diff
@@ -67,43 +65,40 @@ import { NotificationService } from './service/notifications/notification.servic
-      // it throws an error if authority and clientId are undefined, which can be ignored if no auth is needed
-      // the error can be fixed by migrating to version 14, see above
       return {
+        postLoginRoute: '/home',
```
das hier kannst du weglassen

thatsmeta commented on frontend/src/app/auth/auth.service.ts:
```diff
@@ -31,7 +31,7 @@ export class AuthService {
       );
     }
-    return of(true);
+    return of(false);
```
wird diese Methode verwendet? Wenn ja, dann sollte hier true zurückgegeben werden, falls authEnabled false

thatsmeta commented on frontend/src/app/auth/auth.interceptor.ts:
```diff
@@ -30,13 +30,14 @@ export class AuthInterceptor implements HttpInterceptor {
-    if (Utils.isDefined(token)) {
-      request = request.clone({
-        headers: request.headers.set('Authorization', `Bearer ${token}`)
+    if (this.authEnabled && !request.context.get(BYPASS_AUTH)) {
```
```   
if (!this.authEnabled || request.context.get(BYPASS_AUTH)) {
  return next.handle(request);
}
return this.oidcSecurityService.getAccessToken().pipe(
  mergeMap(token => {
              if (Utils.isDefined(token)) {
          request = request.clone({
            headers: request.headers.set('Authorization', `Bearer ${token}`)
          });
        }
        return next.handle(request);
    }  
  )
)
```

pka1024 committed:
angular 16 update

feat(angular 16 update):
- minor review changes.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- updated GitHub action too use node v18.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- updated GitHub action too use node v18.20.2.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- e2e-tests migration.
- auth enabled fixed for the logout menu.

stewit committed:
upgrade node version in shell.nix

stewit committed:
Update nixpkgs version

Commit:
update github action versions

Commit:
fix node version

Commit:
more verbose playwright logging

Commit:
fix playwright target url

pka1024 committed:
angular 16 update

feat(angular 16 update):
- unit-test "http failure response" fixed.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- changed to node version 20.17.0.

Commit:
chore: bump github action versions

pka1024 committed:
angular 16 update

feat(angular 16 update):
- angular Material migration.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- angular Material migration.

pka1024 committed:
angular 16 update

feat(angular 16 update):
- hetida-flowchart packages updated.

Commit:
upgrade node version

pka1024 committed:
angular 16 update

feat(angular 16 update):
- Changed the switch button style.
